llvm/lib/Transforms/Instrumentation/RealtimeSanitizer.cpp

```diff
@@ -45,6 +45,26 @@ static void insertCallAtAllFunctionExitPoints(Function &Fn,
         insertCallBeforeInstruction(Fn, I, InsertFnName);
 }
 
+static PreservedAnalyses rtsanPreservedAnalyses() {
+  PreservedAnalyses PA;
+  PA.preserveSet<CFGAnalyses>();
+  return PA;
```

Comment:
I would get rid of this function and leave these explicit. I think the loop changing we will do will not preserve CFG, so I think it is OK to just do this explicitly in our main block

Reply:
Hmm yeah - part of my refactoring in the main pass function was intended to have a sequence of:

```cpp
if (condition) {
    return transformXYZ(...);
}
else if (condition) {
    return transformABC(...);
}
```

... to express the intent that these transformations are mutually exclusive and return early. But now I'm thinking about it, it doesn't really need to be that way - and it's a bit early days to start those sorts of refactors. So yeah, I think I'll revert this as you suggest

Reply:
On second thoughts, we still need it in two places. I've updated the function name to be more explicit, though - you'll see this in my latest PR. When we no longer need it as part of the `sanitize_realtime` pass, we can inline it then. I think it's better to keep it in this case, don't you?